Mathlib/Data/Fin/Tuple/Basic.lean

```diff
@@ -623,12 +644,49 @@ theorem append_right_eq_snoc {α : Type*} {n : ℕ} (x : Fin n → α) (x₀ : F
     exact (@snoc_last _ (fun _ => α) _ _).symm
 #align fin.append_right_eq_snoc Fin.append_right_eq_snoc
 
+/-- `Fin.snoc` is the same as appending a one-tuple -/
+theorem snoc_eq_append {α : Type*} (xs : Fin n → α) (x : α) :
+    snoc xs x = append xs (cons x Fin.elim0) :=
+  (append_right_eq_snoc xs (cons x Fin.elim0)).symm
+
+theorem append_snoc {n m} {α : Type*} (xs : Fin n → α) (x : α) (ys : Fin m → α) :
```

Comment:
I think this should perhaps be called `append_left_snoc` and the other lemma `append_right_cons`; that matches the `nil` versions.

Reply:
Or alternatively, `snoc_append` and `append_cons` to match `elim0'_append` and `append_elim0'`

Reply:
The infix version would make sense if we have an `HAppend` instance and actually used `++`, we don't have that, though, so I'll go with `left`/`right`.